controller/appcontroller.go

```diff
@@ -2486,11 +2487,46 @@ func (ctrl *ApplicationController) newApplicationInformerAndLister() (cache.Shar
 				oldApp, oldOK := old.(*appv1.Application)
 				newApp, newOK := new.(*appv1.Application)
 				if oldOK && newOK {
+					// Skip refresh requests for informer resync events (same ResourceVersion)
+					if oldApp.ResourceVersion == newApp.ResourceVersion {
+						// Only update sharding and hydration queue, don't trigger refresh
+						if ctrl.hydrator != nil {
+							ctrl.appHydrateQueue.AddRateLimited(newApp.QualifiedName())
+						}
+						ctrl.clusterSharding.UpdateApp(newApp)
+						return
+					}
+
+					// Skip refresh requests for status-only updates (spec unchanged)
+					// This prevents feedback loop: status update → UpdateFunc → requestAppRefresh → refresh → status update
+					if equality.Semantic.DeepEqual(oldApp.Spec, newApp.Spec) {
+						// Check if user requested refresh/hydrate via annotations
+						// Don't skip if user explicitly requested refresh/hydrate
+						oldAnnotations := oldApp.GetAnnotations()
+						newAnnotations := newApp.GetAnnotations()
+						refreshAdded := (oldAnnotations == nil || oldAnnotations[appv1.AnnotationKeyRefresh] == "") &&
+							(newAnnotations != nil && newAnnotations[appv1.AnnotationKeyRefresh] != "")
+						hydrateAdded := (oldAnnotations == nil || oldAnnotations[appv1.AnnotationKeyHydrate] == "") &&
```

Comment:
Is this a new constant `appv1.AnnotationKeyHydrate` added as part of this PR ? I could not find this constant at `./pkg/apis/application/v1alpha1/application_annotations.go` where similar constant `AnnotationKeyRefresh` is defined.